features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```diff
@@ -198,45 +201,112 @@ function APICreateDefault(props) {
+            };
+            restApi.createRevision(api.id, body)
+                .then((api1) => {
+                    const revisionId = api1.body.id;
+                    Alert.info('API Revision created successfully');
+                    setIsRevisioning(false);
+                    const envList = settings.environment.map((env) => env.name);
+                    const body1 = [];
+                    if (envList.length > 0) {
+                        body1.push({
+                            name: envList[0],
+                            displayOnDevportal: true,
+                        });
+                    }
+                    // for (let i = 0; i < envList.length; i++) {
+                    //     body1.push({
+                    //         name: envList[i],
+                    //         displayOnDevportal: true,
+                    //     });
+                    // }
+                    setIsDeploying(true);
+                    restApi.deployRevision(api.id, revisionId, body1)
+                        .then(() => {
+                            Alert.info('API Revision Deployed Successfully');
+                            setIsDeploying(false);
+                        })
+                        .catch((error) => {
+                            if (error.response) {
+                                Alert.error(error.response.body.description);
+                                setPageError(error.response.body);
+                            } else {
+                                const message = 'Something went wrong while deploying the API Revision';
+                                Alert.error(intl.formatMessage({
+                                    id: 'Apis.Create.Default.APICreateDefault.error.errorMessage.deploy.revision',
+                                    defaultMessage: message,
+                                }));
+                                setPageError(message);
+                            }
+                            console.error(error);
+                        })
+                        .finally(() => {
+                            setIsDeploying(false);
+                        });
+                })
+                .catch((error) => {
+                    if (error.response) {
+                        Alert.error(error.response.body.description);
+                        setPageError(error.response.body);
+                    } else {
+                        const message = 'Something went wrong while creating the API Revision';
+                        Alert.error(intl.formatMessage({
+                            id: 'Apis.Create.Default.APICreateDefault.error.errorMessage.create.revision',
+                            defaultMessage: message,
+                        }));
+                        setPageError(message);
+                    }
+                    console.error(error);
+                })
+                .finally(() => {
+                    setIsRevisioning(false);
```

Comment:
I think we should throw an exception here and break the execution. Prevent it from publishing and deploying